main.tf

```diff
@@ -17,11 +15,11 @@ resource "null_resource" "provisioner" {
 
   triggers {
     signature = "${data.archive_file.default.output_md5}"
-    command   = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
+    command   = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", compact(var.arguments))} ${length(compact(var.envs)) > 0 ? "-e" : ""} ${length(compact(var.envs)) > 0 ? join(" -e ", var.envs) : ""} ${var.playbook}"
   }
 
   provisioner "local-exec" {
-    command = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
+    command = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", compact(var.arguments))} ${length(compact(var.envs)) > 0 ? "-e" : ""} ${length(compact(var.envs)) > 0 ? join(" -e ", var.envs) : ""} ${var.playbook}"
```

Comment:
`${length(compact(var.envs)) > 0 ? join(" -e ", var.envs) : ""}`
should be
`${join(" -e ", compact(var.envs))}`

Reply:
@comeanother , in this case will get the error with empty `var.envs`
```sh
ansible-playbook: error: -e option requires an argument
```

Reply:
What would be if `var.envs` has empty elements like `""`?
I assume, `ansible-playbook` command will fail, because it will be like:
```
ansible-playbook -e -e var=value
```

Reply:
look at check condition`length(compact(var.envs))`

Reply:
Let's consider the following example: 
```
variable envs {
  default = ["", "TEST=test"]
}
```
It will pass condition check `length(compact(var.envs))`. but fail later
```
ansible-playbook -e -e TEST=test
```

Reply:
We can't prevent all misstakes. If we add compact() for var.envs - using some vars like "dev=name title" will be impossible.

Reply:
@osterman, what do you think about that?

Reply:
For examole if use some vars like "name=${some_words}" it'll break module too :) . What check should be added ? :)

Reply:
`${join(" -e ", compact(var.envs))}` works properly.

Example:
```
variable "envs" {
  default = ["", "test"]
}

variable "dry_run" {
  default = "false"
}

variable "arguments" {
  default = []
}

variable "playbook" {
  default = ""
}

output "b" {
  value = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", compact(var.arguments))} ${length(compact(var.envs)) > 0 ? "-e" : ""} ${length(compact(var.envs)) > 0 ? join(" -e ", var.envs) : ""} ${var.playbook}"
}

output "c" {
  value = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", compact(var.arguments))} ${length(compact(var.envs)) > 0 ? "-e" : ""} ${join(" -e ", compact(var.envs))} ${var.playbook}"
}
```

![screenshot from 2017-10-13 15-33-32](https://user-images.githubusercontent.com/11299538/31546438-fcf5e576-b02b-11e7-809a-5b3e4ef96806.png)